gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -49,10 +49,67 @@ DEFAULT_CLIENT_INFO = gapic_v1.client_info.ClientInfo(
     rest_version=requests_version,
 )
 
+
+class {{ service.name }}RestInterceptor:
```

Comment:
Is the expectation that we subclass this and override the `pre_*` and `post_*` methods below? Or I suppose we could pass an arbitrary object that has the necessary methods.

Reply:
Since python is duck typed, so if the transport is passed something that has the appropriate `pre_*` and `post_*` methods the actual type doesn't matter. I just added this generated class to make it easy to selectively intercept particular calls and have a reasonable, i.e. do nothing, default.